### PR TITLE
feat: allow user to set TOC display preference

### LINKED ISSUE #6383: Add display preference to move document TOC to right hand side
Currently thinking this would be a workspace level preference.

### PR DESCRIPTION
**Pull request by hmacr**
Closes #6383

### CODE REVIEW

File: app/scenes/Settings/Details.tsx [hunk 1]
```diff
@@ -276,6 +282,31 @@ function Details() {
             />
           </SettingRow>
 
+          <SettingRow
```
Comment (tommoor) [inline app/scenes/Settings/Details.tsx:285]: Lets move this under the "Display" header, it's currently under "Behavior"

File: app/scenes/Settings/Details.tsx [hunk 2]
```diff
@@ -276,6 +282,31 @@ function Details() {
...
+            border={false}
+            label={t("Table of Contents Position")}
```
Comment (tommoor) [inline app/scenes/Settings/Details.tsx:287]:
```suggestion
            label={t("Table of Contents position")}
```

File: app/scenes/Settings/Details.tsx [hunk 3]
```diff
@@ -276,6 +282,31 @@ function Details() {
...
+            name="tocPosition"
+            description={t(
+              "The side to display the table of contents in relation to the main content."
+            )}
+          >
+            <InputSelect
+              ariaLabel={t("TOC position")}
```
Comment (tommoor) [inline app/scenes/Settings/Details.tsx:294]:
```suggestion
              ariaLabel={t("Table of Contents position")}
```

File: app/scenes/Settings/Details.tsx [hunk 4]
```diff
@@ -276,6 +282,31 @@ function Details() {
...
+              options={[
+                {
+                  label: "Left",
```
Comment (tommoor) [inline app/scenes/Settings/Details.tsx:297]:
```suggestion
                  label: t("Left"),
```

File: app/scenes/Settings/Details.tsx [hunk 5]
```diff
@@ -276,6 +282,31 @@ function Details() {
...
+                  value: TOCPosition.Left,
+                },
+                {
+                  label: "Right",
```
Comment (tommoor) [inline app/scenes/Settings/Details.tsx:301]:
```suggestion
                  label: t("Right"),
```

File: app/scenes/Settings/Details.tsx [hunk 6]
```diff
@@ -276,6 +282,31 @@ function Details() {
...
+                  value: TOCPosition.Right,
+                },
+              ]}
+              value={tocPosition}
+              onChange={(p: TOCPosition) => setTocPosition(p)}
```
Comment (tommoor) [inline app/scenes/Settings/Details.tsx:306]:
```suggestion
              onChange={setTocPosition}
```

Should work, no? Or is typescript forcing the intermediate function

File: app/scenes/Document/components/Document.tsx [hunk 7]
```diff
@@ -540,6 +542,7 @@ class DocumentScene extends React.Component<Props> {
                         <Contents
                           headings={this.headings}
                           isFullWidth={document.fullWidth}
+                          onLeftSide={tocOnLeftSide}
```
Comment (tommoor) [inline app/scenes/Document/components/Document.tsx:545]:
```suggestion
                          position={position}
```

Lets continue to pass `position` down

**Review by tommoor — COMMENTED** [review]
This is a good start! The right side position does have some layout issues though and easily gets obscured by having the comment or history sidebar open.

File: app/scenes/Settings/Details.tsx (hunk 6 above)
Comment (hmacr) [reply app/scenes/Settings/Details.tsx:306]: Yeah, Typescript 😅 .

**Comment by hmacr** [comment]
> This is a good start! The right side position does have some layout issues though and easily gets obscured by having the comment or history sidebar open.

Ohh.. Sorry, I missed testing the layout shifts with the sidebar open.
Let me update the code.

**Comment by tommoor** [comment]
Give it a go, honestly how the sidebar lives in the dom might just need rethinking – it's not especially great on the left side either :(

**Comment by hmacr** [comment]
> This is a good start! The right side position does have some layout issues though and easily gets obscured by having the comment or history sidebar open.

When the sidebar is open, I don't find the TOC being obscured; however I do find layout shifts, no matter the display preference.
Let me redesign the document layout to fix them.

**Comment by almereyda** [comment]
This seems interesting. Does it require additional attention?

**Comment by hmacr** [comment]
> This seems interesting. Does it require additional attention?

@almereyda Sorry, I got around to it a couple of days ago after a break.

**Comment by hmacr** [comment]
Created another PR #6943 as the git tree was messed up for this branch.

**Pull request closed without merging**